### PR TITLE
Changes for PAM support needed on RHEL7/Centos7

### PR DESCRIPTION
**Pull request by WetHippie**
The following change adds in support for the changed path and filename for the PAM module. This can be used if you have PAM already configured to support LDAP rather than hitting up LDAP directly (ie UsePAM => true, rather than UseLDAP => true). ie this is another way of tackling the problems raised in #172

### CODE REVIEW

**Comment by luxflux** [comment]
Thank you for sorting this out.
I tried to resolve the merge conflicts, could you verify the functionality again?

**Comment by WetHippie** [comment]
Can do. Give me a day or so to work through it all with the tests. Should be simple, just busy at work today.

**Comment by WetHippie** [comment]
I can't get the tests running as currently you explicitly require puppet-lint 1.0.1 in the Gemfile. Current lint is 2.1.0. Could I upgrade the Gem to that and work on updating the tests?

**Comment by luxflux** [comment]
> I can't get the tests running as currently you explicitly require puppet-lint 1.0.1 in the Gemfile. Current lint is 2.1.0. Could I upgrade the Gem to that and work on updating the tests?

Sure, feel free to update it.

**Comment by WetHippie** [comment]
Lint updated. Lots of changes to be 4.x compatible. Now the tests are completely broken :( I'll have to dig through and find out just what it is up as they seem to be broken in some really fundamental ways - like not even being able to instance the core class due to missing global variables like ::osfamily. Might be a few more days yet.

**Comment by wyardley** [comment]
@WetHippie Do you want to rebase (and squash) this for review?

File: manifests/ca.pp [hunk 1]
```diff
@@ -100,13 +100,13 @@
   $ssl_key_size = 1024,
   $ca_expire    = 3650,
   $key_expire   = 3650,
-  $key_cn       = '',
-  $key_name     = '',
-  $key_ou       = '',
+  $key_cn       = undef,
+  $key_name     = undef,
+  $key_ou       = undef,
```
Comment (wyardley) [inline manifests/ca.pp]: I like this change

File: manifests/ca.pp [hunk 2]
```diff
@@ -100,13 +100,13 @@
...
   $tls_auth     = false,
 ) {
 
-  include openvpn
+  include ::openvpn
```
Comment (wyardley) [inline manifests/ca.pp]: With puppet4+, the absolute path check is no longer needed, and in fact, `include foo` is preferred over `include ::foo`

File: manifests/client.pp [hunk 3]
```diff
@@ -349,80 +349,80 @@
...
     content => "<key>\n",
-    order   => '05'
+    order   => '05',
   }
 
   concat::fragment { "${etc_directory}/openvpn/${server}/download-configs/${name}.ovpn/key":
     target => "${etc_directory}/openvpn/${server}/download-configs/${name}.ovpn",
     source => "${etc_directory}/openvpn/${server}/download-configs/${name}/keys/${name}/${name}.key",
-    order  => '06'
+    order  => '06',
```
Comment (wyardley) [inline manifests/client.pp]: These should now be updated w/ the switch to Vox

**Comment by bastelfreak** [comment]
Hi @WetHippie, are you still interested in getting this merged? If so, could you please rebase?

**Comment by bastelfreak** [comment]
Hi. I'm going to close this PR due to inactivity. Please rebase it and address the comments above if you're still interested in it.

**Pull request closed without merging**